lib/actions/action-maintain-contact.ts

```diff
@@ -32,17 +36,24 @@ const handler: ActionFile['handler'] = async (
 		'contact@1.0.0',
 	)) as TypeContract;
 	assert.INTERNAL(
-		request.context,
+		request.logContext,
 		typeCard,
-		context.errors.WorkerNoElement as any, // TS-TODO fix this
+		workerErrors.WorkerNoElement,
 		'No such type: contact',
 	);
 
-	const originCard =
-		card.data.origin &&
-		(isUUID(card.data.origin)
+	// Get origin and source
+	let origin = '';
+	let source = '';
+	if (typeof card.data.origin === 'string') {
+		origin = card.data.origin;
+		const originCard = isUUID(card.data.origin)
 			? await context.getCardById(session, card.data.origin)
-			: await context.getCardBySlug(session, card.data.origin));
+			: await context.getCardBySlug(session, card.data.origin);
+		if (originCard && typeof originCard.data.source === 'string') {
+			source = originCard.data.source;
+		}
+	}
```

Comment:
@Ereski A bit verbose, but trying to get away from `card.data.origin as string`. What do you think?

Reply:
So we just ignore if `card.data.origin` is of the wrong type or doesn't exist?

Reply:
That was the idea. Did some testing and found that this does change the end result so I think I'll revert. I'll go back to `as string` with a `TS-TODO`. I'd rather not change return values in this PR if possible.